components/domains/externalDomainTransferModal.tsx

```diff
@@ -0,0 +1,176 @@
+import { isHexString, minifyAddress } from "../../utils/stringService";
+import Button from "../UI/button";
+import { utils } from "starknetid.js";
+import { StarknetIdJsContext } from "../../context/StarknetIdJsProvider";
+import ConfirmationTx from "../UI/confirmationTx";
+import { useNotificationManager } from "../../hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "../../utils/constants";
+import { Call } from "starknet";
+
+type ExternalDomainsTransferModalProps = {
+  domain: string;
+  domainEncoded: string;
+  resolverContract: string;
+  handleClose: () => void;
+  isModalOpen: boolean;
+  domainKind: DomainKind;
+};
+
+const ExternalDomainsTransferModal: FunctionComponent<
+  ExternalDomainsTransferModalProps
+> = ({
+  domainEncoded,
+  resolverContract,
+  handleClose,
+  isModalOpen,
+  domainKind,
+}) => {
+  const router = useRouter();
+  const { externalDomain: domain } = router.query;
+  const [targetAddress, setTargetAddress] = useState<string>("");
+  const { addTransaction } = useNotificationManager();
+  const [addressInput, setAddressInput] = useState<string>("");
+  const { starknetIdNavigator } = useContext(StarknetIdJsContext);
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [callData, setCallData] = useState<Call[]>([]);
+
+  const { writeAsync: transfer_name, data: transferData } = useContractWrite({
+    calls: callData,
+  });
+
+  useEffect(() => {
+    setCallData([
+      {
+        contractAddress: resolverContract,
+        entrypoint: "transfer_name",
+        calldata: [domainEncoded, targetAddress],
+      },
+    ]);
+  }, [domainEncoded, targetAddress, resolverContract]);
+
+  useEffect(() => {
+    if (isHexString(addressInput)) {
+      setTargetAddress(addressInput);
+    } else if (utils.isStarkDomain(addressInput)) {
+      starknetIdNavigator?.getAddressFromStarkName(addressInput).then((res) => {
+        if (!res || res === "0x0") setTargetAddress("");
+        else setTargetAddress(res);
+      });
+    } else {
+      setTargetAddress("");
+    }
+  }, [addressInput, starknetIdNavigator]);
+
+  useEffect(() => {
+    if (!transferData?.transaction_hash) return;
+    addTransaction({
+      timestamp: Date.now(),
+      subtext: `${domain} transfered`,
+      type: NotificationType.TRANSACTION,
+      data: {
+        type: TransactionType.TRANSFER_EXTERNAL_DOMAIN,
+        hash: transferData.transaction_hash,
+        status: "pending",
+      },
+    });
+    setIsTxSent(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [transferData]); // We want to call this only once when the transaction is sent
+
+  function transferName(): void {
+    transfer_name();
+  }
+
+  function changeAddress(value: string): void {
+    setAddressInput(value);
+  }
+
+  function closeModal(): void {
+    setIsTxSent(false);
+    handleClose();
+  }
+
+  function getResolverCondition(domainKind: DomainKind): string | undefined {
```

Comment:
I added a comment here also to move that to utils + test it. Did you forget or didn't do it on purpose thinking it was better for it to stay here ?

Reply:
As I was saying in this comment, I updated the function : https://github.com/starknet-id/app.starknet.id/pull/748#discussion_r1581029618 
I can move it, but I'm not sure it will be useful elsewhere as it returns a text which is really specific to this component